tools/generator-terraform/generator/definitions/template_service_client_test.go

```diff
@@ -43,10 +45,95 @@ func NewClient(o *common.ClientOptions) (*AutoClient, error) {
+	actual := templateForServiceClient(input)
+	expected := `
+package client
+
+import (
+	resourcesV20151101Preview "github.com/hashicorp/go-azure-sdk/resource-manager/resources/2015-11-01-preview"
+	resourcesV20200101 "github.com/hashicorp/go-azure-sdk/resource-manager/resources/2020-01-01"
+	"github.com/hashicorp/terraform-provider-myprovider/internal/common"
+)
+
+type AutoClient struct {
+	V20151101Preview resourcesV20151101Preview.Client
+	V20200101 resourcesV20200101.Client
+}
+
+func NewClient(o *common.ClientOptions) (*AutoClient, error) {
+	v20151101PreviewClient, err := resourcesV20151101Preview.NewClientWithBaseURI(o.Environment.ResourceManager, func(c *resourcemanager.Client) {
+		o.Configure(c, o.Authorizers.ResourceManager)
+	})
+	if err != nil {
+		return nil, fmt.Errorf("building client for resources v20151101Preview: %+v", err)
+	}
+	v20200101Client, err := resourcesV20200101.NewClientWithBaseURI(o.Environment.ResourceManager, func(c *resourcemanager.Client) {
+		o.Configure(c, o.Authorizers.ResourceManager)
+	})
+	if err != nil {
+		return nil, fmt.Errorf("building client for resources v20200101: %+v", err)
+	}
+	return &AutoClient{
+		V20151101Preview: v20151101PreviewClient,
+		V20200101: v20200101Client,
+	}, nil
+}
+`
+	testhelpers.AssertTemplatedCodeMatches(t, expected, actual)
+}
+
 func TestTemplateForServiceClient_GoAutoRest(t *testing.T) {
 	input := models.ServiceInput{
-		ApiVersion:         "2015-11-01-preview",
-		ProviderPrefix:     "myprovider", // intentionally not used, since this is `client`
+		ProviderPrefix: "myprovider", // intentionally not used, since this is `client`
+		ResourceToApiVersion: map[string]string{
+			"load_test_service_resource": "2015-11-01-preview",
+		},
+		SdkServiceName:     "loadtestservice",
+		ServicePackageName: "mypackage",
+	}
+	actual := templateForServiceClient(input)
+	expected := `
+package client
+
+import (
+	"github.com/Azure/go-autorest/autorest"
+	loadtestserviceV20151101Preview "github.com/hashicorp/go-azure-sdk/resource-manager/loadtestservice/2015-11-01-preview"
+	"github.com/hashicorp/terraform-provider-myprovider/internal/common"
+)
+
+type AutoClient struct {
+	V20151101Preview loadtestserviceV20151101Preview.Client
+	}
+
+func NewClient(o *common.ClientOptions) (*AutoClient, error) {
+	v20151101PreviewClient := loadtestserviceV20151101Preview.NewClientWithBaseURI(o.ResourceManagerEndpoint, func(c *autorest.Client) {
+		c.Authorizer = o.ResourceManagerAuthorizer
+	})
+	return &AutoClient{
+		V20151101Preview: v20151101PreviewClient,
+	}, nil
+}
+`
+	testhelpers.AssertTemplatedCodeMatches(t, expected, actual)
+}
+
+func TestTemplateForServiceClient_GoAutoRest_WithMultupleApiVersions(t *testing.T) {
+	input := models.ServiceInput{
+		ProviderPrefix: "myprovider", // intentionally not used, since this is `client`
+		ResourceToApiVersion: map[string]string{
+			"load_test_service_one_resource": "2015-11-01-preview",
+			"load_test_service_two_resource": "2021-10-10",
```

Comment:
can we add another test with the same API version defined in multiple resources? e.g. 2 using `2015-11-01-preview` and one using `2021-10-10`, to ensure we unique these before outputting them?